From: Simon McVittie <smcv@collabora.com>
Date: Fri, 16 Apr 2021 10:54:50 +0100
Subject: oci-registry: Fix error reporting

The intention here seems to have been that failing to close the http
stream provokes a warning but does not make the function fail, but we
were setting the wrong error, resulting in a NULL dereference if closing
the http stream somehow fails.

Signed-off-by: Simon McVittie <smcv@collabora.com>
Forwarded: https://github.com/flatpak/flatpak/pull/4228
---
 common/flatpak-oci-registry.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/flatpak-oci-registry.c b/common/flatpak-oci-registry.c
index 2a7f3ba..2dbd46b 100644
--- a/common/flatpak-oci-registry.c
+++ b/common/flatpak-oci-registry.c
@@ -2829,7 +2829,7 @@ load_oci_index (GFile        *index,
   if (json == NULL)
     return NULL;
 
-  if (!g_input_stream_close (G_INPUT_STREAM (in), cancellable, error))
+  if (!g_input_stream_close (G_INPUT_STREAM (in), cancellable, &local_error))
     g_warning ("Error closing http stream: %s", local_error->message);
 
   return (FlatpakOciIndexResponse *) g_steal_pointer (&json);
